### PR TITLE
Single Column Model Updates & Fixes for v3

### PR DESCRIPTION
**Pull request by bogensch**
Modify and fix the single column model (SCM) to make it compatible with master and v3.  Specifically: 

1. The config_component for SCM_EAM has been changed to “scam_generic” (which should have never been changed).  When running cases with the standardized scripts prescribed aerosols are used, which is not compatible with the scm_generic_chemUCI-Linoz-mam5-vbs component currently specified.  (Aside: a new prescribed aerosol file has been generated by running E3SM with mam5.  This file has been uploaded to the E3SM data server and the E3SM SCM scripts have been updated to use this file).
2. Idealization flags have been added to the P3 microphysics to allow for:
    1. Runs with no precipitation.  Note that in the P3 implementation this only turns off liquid precipitation, which is used in several GCSS boundary layer cloud cases in the E3SM SCM library.  
    2. Prescribed droplet concentration.  This was already a feature in place in P3 but this PR connects this functionality to the EAM namelist.
3. Fixes to allow the ability to “replay” a column once again (back by semi-popular demand!).  This was a feature that worked in E3SMv1 but has not been functional for sometime.  The fix in this PR allows users to generate output needed to make IOP forcing files to replay a single column.  Note that additional post-processing is required to actually do an SCM run (in which the user needs to consult with the E3SM wiki page).  

Further documentation and scripts have been updated on the E3SM SCM wiki to reflect changes made in this PR (https://github.com/E3SM-Project/scmlib/wiki/E3SM-Single-Column-Model-Home).

[BFB]

All e3sm_developer tests pass on pm-cpu.

### CODE REVIEW

**Comment by rljacob** [comment]
@brhillman please review.

**Comment by rljacob** [comment]
@brhillman please review this

**Review by brhillman — APPROVED** [review]

File: components/eam/cime_config/config_component.xml [hunk 1]
```diff
@@ -136,7 +136,7 @@
       <value compset="20TR(?:SOI)?_EAM.*CHEMUCI.*LINOZ.*MAM5"  >20TR_eam_chemUCI-Linoz-mam5</value>
       <value compset="ARM95_EAM"                        >scam_arm95</value>
       <value compset="ARM97_EAM"                        >scm_arm97_chemUCI-Linoz-mam5-vbs</value>
-      <value compset="SCM_EAM"                          >scm_generic_chemUCI-Linoz-mam5-vbs</value>
+      <value compset="SCM_EAM"                          >scam_generic</value>
```
Comment (rljacob) [inline components/eam/cime_config/config_component.xml:139]: @bogensch should we add a test of this compset?  We only test ARM97_EAM

Comment (bogensch) [reply components/eam/cime_config/config_component.xml:139]: Yes, it's probably wise.  I'll try to get this done before I leave for vacation tomorrow afternoon... else expect a period of radio silence (FYI).

Comment (wlin7) [reply components/eam/cime_config/config_component.xml:139]: Will need a new testmod for it, to bring in some of the settings specified in SCM case scripts to allow setup to proceed. (e.g., scmlat, scmlon)

**Comment by wlin7** [comment]
Hi @bogensch , thanks for working to update the SCM configuration. I didn't realize when putting in v3atm that SCM run scripts rely on the generic use_case file. I modified the use_case for the nightly ARM97 test, and without careful thinking, also applied the changes to the generic one. As Rob suggested, a test using scam_generic will be useful for protecting this configuration. It can replace the other test (FSCM-ARM97) or revert the change in its use_case file and associated config option to be more like scam_generic.

Unrelated to this PR itself: I found the link to the 'desired cases' on the instruction for '[Running the single column model](https://github.com/E3SM-Project/scmlib/wiki/Running-the-E3SM-Single-Column-Model)' returns a page that I didn't expect: It 
 would be to Create a new page. I am not sure if this is intended and how it works if so.
 
 Also have a few comments on the run script, if you have time to update them in the near future.
- Can replace machine cori with pm-cpu, and $CSCRATCH with $PSCRATCH
- [Line 198 of the runscript](https://github.com/E3SM-Project/scmlib/blob/master/E3SM_SCM_scripts/run_e3sm_scm_ARM97.csh#L198 ) has a duplicated keyword 'set'
- a number of if-logics are conditioned on if the code version is 'master' . Users may not fully comprehend the distinction you try to convey. Maybe some like 'v3x' to use either master or any variant branch a user creates?
- namelist parameter values in the script going to user_nl_eam. Right now there is no distinction between v2 and v3.  Most of them should have been set by default, given v2 or v3 code base. Can we remove those that are not SCM specific?

**Review by wlin7 — APPROVED** [review]
The PR itself looks good. Verified atm_developer tests remain BFB.

**Comment by rljacob** [comment]
@wlin7 you can start merging this if you want.  We could wait and add the test in a later PR.

**Comment by bogensch** [comment]
@wlin7 thanks for your careful review... not only of the code but for the scripts.  I will incorporate your great suggestions to the scripts.

**Comment by wlin7** [comment]
Thank you, @bogensch . As the scripts are not part of this PR, I can start merging today if next is open, while you work on updating the run scripts.

**Comment by wlin7** [comment]
Merged to next.

**Pull request merged**